content/tutorials/how-to-send-l1-l2-transaction/10.index.md

```diff
@@ -0,0 +1,335 @@
+transaction on L1 and executing the transaction on L2 using
+`getBaseCost` from the `Wallet` class. This method:
+
+- Retrieves the ZKsync BridgeHub contract address calling `zks_getBridgehubContract` RPC method.
+- Calls the `l2TransactionBaseCost` function on the ZKsync BridgeHub system contract to retrieve the fee.
+
+  ```ts
+  const baseCost = await wallet.getBaseCost({
+    // L2 computation
+    gasLimit: l2GasLimit,
+    // L1 gas price
+    gasPrice: l1GasPrice,
+  });
+
+  console.log(`Executing this transaction will cost ${ethers.utils.formatEther(baseCost)} ETH`);
+  ```
+
+1. Encode the transaction calldata:
+
+  ```ts
+  const iface = new ethers.utils.Interface(ABI);
+  const calldata = iface.encodeFunctionData("setGreeting", [message]);
+  ```
+
+1. Finally, send the transaction from your wallet using the `requestExecute` method. This helper method:
+
+- Retrieves the Zksync BridgeHub contract address calling `zks_getBridgehubContract`.
+- Populates the L1-L2 transaction.
+- Sends the transaction to the `requestL2TransactionDirect` function of the ZKsync BridgeHub system contract on L1.
+
+  ```ts
+  const txReceipt = await wallet.requestExecute({
+    // destination contract in L2
+    contractAddress: L2_CONTRACT_ADDRESS,
+    calldata,
+    l2GasLimit: l2GasLimit,
+    refundRecipient: wallet.address,
+    overrides: {
+      // send the required amount of ETH
+      value: baseCost,
+      gasPrice: l1GasPrice,
+    },
+  });
+  txReceipt.wait();
+  ```
+
+## Full example
+
+Create a `send-l1-l2-tx.ts` file in the root directory with the next script:
+
+::drop-panel
+  ::panel{label="send-l1-l2-tx.ts"}
+
+  ```ts [send-l1-l2-tx.ts]
+    import { Contract, Wallet, Provider, types } from "zksync-ethers";
+    import * as ethers from "ethers";
+
+    // load env file
+    import dotenv from "dotenv";
+    dotenv.config();
+
+    // Greeter contract ABI for example
+    const ABI = [
+      {
+        inputs: [
+          {
+            internalType: "string",
+            name: "_greeting",
+            type: "string",
+          },
+        ],
+        stateMutability: "nonpayable",
+        type: "constructor",
+      },
+      {
+        inputs: [],
+        name: "greet",
+        outputs: [
+          {
+            internalType: "string",
+            name: "",
+            type: "string",
+          },
+        ],
+        stateMutability: "view",
+        type: "function",
+      },
+      {
+        inputs: [
+          {
+            internalType: "string",
+            name: "_greeting",
+            type: "string",
+          },
+        ],
+        name: "setGreeting",
+        outputs: [],
+        stateMutability: "nonpayable",
+        type: "function",
+      },
+    ];
+
+    // RPC endpoints
+    const L1_RPC_ENDPOINT = "https://rpc2.sepolia.org";
+    const L2_RPC_ENDPOINT = "https://sepolia.era.zksync.dev";
+
+    const WALLET_PRIV_KEY = process.env.WALLET_PRIV_KEY || "";
```

Comment:
this doesn't match the .env file `WALLET_PRIVATE_KEY=0x..;`

Reply:
fixed